### PR TITLE
Fix overflow issues

### PR DESCRIPTION
**Pull request by njokuScript**
## Description

This PR addresses issues in the LaunchboxExchange contract focusing on preventing arithmetic overflow in the `getAmountOutWithFee` function.

### Changes

1. Updated `getAmountOutWithFee` function
- Changed fee representation to basis points (1/10000) for more precise fee settings.
- Implemented a safe `mulDiv` function to prevent overflow in calculations.
- Updated input validation for trade fee.

2. Constants and state variables
- Added `FEE_DENOMINATOR` constant set to 10000.

### Rationale
The previous implementation was susceptible to arithmetic overflow, especially with high fee values. The new implementation uses a more robust calculation method and allows for more precise fee settings.

### Impact

- `tradeFee` is now represented in basis points. Existing fee values need to be updated accordingly (e.g., 10% fee should now be set to 1000).
- The contract can now handle higher fee values without overflow issues.

### CODE REVIEW

File: src/launchbox/exchange/LaunchboxExchange.sol [hunk 1]
```diff
@@ -118,13 +124,82 @@ contract LaunchboxExchange {
...
+
+        // Calculate fee
+        fee = mulDiv(amountIn, _tradeFee, FEE_DENOMINATOR);
+
+        // Calculate amountInWithFee
+        uint256 amountInWithFee = amountIn - fee;
+
+        // Calculate numerator and denominator
+        uint256 numerator = mulDiv(amountInWithFee, reserveOut, 1);
+        uint256 denominator = reserveIn * FEE_DENOMINATOR + amountInWithFee;
```
Comment (codebuster22) [inline src/launchbox/exchange/LaunchboxExchange.sol:135]: This changes the Math.

earlier the formula was: 

```math
\frac{\text{amountIn} \times (\text{FEE\_DENOMINATOR} - \text{\_tradeFee}) \times \text{reserveOut}}{(\text{reserveIn} \times \text{FEE\_DENOMINATOR}) + (\text{amountIn} \times (\text{FEE\_DENOMINATOR} - \text{\_tradeFee}))}
```

But now it is:
```math
\frac{\left( \text{amountIn} \times \left( \text{FEE\_DENOMINATOR} - \text{\_tradeFee} \right) / \text{FEE\_DENOMINATOR} \right) \times \text{reserveOut}}{\left( \text{reserveIn} \times \text{FEE\_DENOMINATOR} \right) + \left( \text{amountIn} \times \left( \text{FEE\_DENOMINATOR} - \text{\_tradeFee} \right) \right)}
```

Comment (njokuScript) [reply src/launchbox/exchange/LaunchboxExchange.sol:135]: let me update it

Comment (njokuScript) [reply src/launchbox/exchange/LaunchboxExchange.sol:135]: done

**Review by codebuster22 — APPROVED** [review]
looks good. We can work with this. Though, dividing at the end would give better precision, but it's fine.

**Pull request merged**